src/WarcraftLegacies.Source/Quests/Warsong/QuestWarsongOutpost.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Warsong
+{
+  /// <summary>
+  /// The Warsong clan can acquire the Warsong Outpost in Uldum.
+  /// </summary>
+  public sealed class QuestWarsongOutpost : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+    private readonly ObjectiveAnyUnitInRect _enterWarsongOutpostRegion;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestWarsongOutpost"/>.
+    /// </summary>
+    public QuestWarsongOutpost() : base("Warsong Outpost in Uldum",
+      "The desert of Uldum is rumoured to conceal a powerful weapon. We must send one of our heroes to claim leadership of the outpost and explore further.",
```

Comment:
It's honestly quite good, and I like that you used Uldum specific lore. However I think the "powerful weapon" bit implies we actually have something like that in the game, and we don't.
```suggestion
      "The deserts of Uldum are littered with ancient debris from a lost age, and it seems some of its secrets remain intact even now.  This matters little to the Warsong, however; this land is merely another target ripe for conquest.",
```